pkg/httpapi/api.go

```diff
@@ -226,6 +227,80 @@ func (a *APIRouter) ListApplications(w http.ResponseWriter, r *http.Request) {
 	marshalResponse(w, applicationsToAppsResponse(apps, parsedRepoURL.String()))
 }
 
+func (a *APIRouter) GetApplicationHistory(w http.ResponseWriter, r *http.Request) {
+	params := httprouter.ParamsFromContext(r.Context())
+	envName, appName := params.ByName("env"), params.ByName("app")
+	app := &argoV1aplha1.Application{}
+
+	repoURL := strings.TrimSpace(r.URL.Query().Get("url"))
+	if repoURL == "" {
+		log.Println("ERROR: please provide a valid GitOps repo URL")
+		http.Error(w, "please provide a valid GitOps repo URL", http.StatusBadRequest)
+		return
+	}
+
+	parsedRepoURL, err := url.Parse(repoURL)
+	if err != nil {
+		log.Printf("ERROR: failed to parse URL, error: %v", err)
+		http.Error(w, fmt.Sprintf("failed to parse URL, error: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	parsedRepoURL.RawQuery = ""
+
+	appList := &argoV1aplha1.ApplicationList{}
+	var listOptions []ctrlclient.ListOption
+
+	listOptions = append(listOptions, ctrlclient.InNamespace(""), ctrlclient.MatchingFields{
+		"metadata.name": fmt.Sprintf("%s-%s", envName, appName),
+	})
+
+	err = a.k8sClient.List(r.Context(), appList, listOptions...)
```

Comment:
Since there will be only one application with the given name, should we use `Get()` instead of `List()`?

Reply:
Hi @chetan-rns , I think Get() will require knowing the namespace in the `ObjectKey`.  I think we will need to hardcode 'openshift-gitops'.   The List call uses the `listOptions` to get the name from the metadata. I think only one app is returned here in the list.  Also this is similar to how the other API, `GetApplicationDetails` is done.  We use `envName`, `appName` and the gitRepoURL to identify the app.

Note, if we are are go make this work generically, without kam, we probably will need to use the app name, the namespace, and maybe the presence of a label that the user (or kam) can add, in order to make the apps show up in the environments pages.   Perhaps we should make these changes then.  What do you think?